src/main/java/science/icebreaker/service/JwtTokenValidationService.java

```diff
@@ -1,36 +1,77 @@
 package science.icebreaker.service;
 
 import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
 import io.jsonwebtoken.Jwts;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+
 import science.icebreaker.dao.entity.Account;
 
 
 @Service
 public class JwtTokenValidationService {
 
     private final String jwtSecret;
+    private final Set<String> invalidatedTokens;
```

Comment:
Is this per session or for all sessions ever? Or asked differently: when will this set (and the map) get discarded?

Reply:
The data structures are for all sessions. I can add periodic checks to discard already expired tokens from the set and remove any entry from the map which has a time less than (currentTime - tokenLifeTime) but I don't think this would be a very useful use case for now since the ultimate goal is to implement the token store in some sort of persistent store, not in the process memory. I can decouple the token store to facilitate extension in the future.

Note: if we are expecting a relatively large number of logouts, It would make sense to implement this, even for the in memory store.

Reply:
Please upon an issue for that - in the long run it will be important to discard outdated tokens. It could be used as attack to have the server run out of memory, I think.

Reply:
An important note is that this comes under rate limits. An attacker with a valid account can still create a huge number of valid tokens in a short time, and removing tokens periodically when expired won't fix this.